api/lib/controllers/avatar.js

```diff
@@ -0,0 +1,38 @@
+const Avatar = require("../models/avatar");
+
+module.exports = {
+  get: async (req, res) => {
+    const subject_id = req.credentials.sub;
+    const avatar = await Avatar.findOne({ owner_id: subject_id });
+
+    res.status(200).send(avatar);
+  },
+
+  create: async (req, res) => {
+    const subject_username = req.credentials.user.username;
+    const subject_id = req.credentials.sub;
+
+    const getColor = () => {
+      return (
+        "hsl(" +
+        360 * Math.random() +
+        "," +
+        '100%,50%)'
+      );
+    };
+    const initials = subject_username
+      .match(/(\b\S)?/g)
+      .join("")
+      .match(/(^\S|\S$)?/g)
+      .join("")
+      .toUpperCase();
```

Comment:
Since the `initials` are calculated based on the `user.username` we shouldn't store them. Otherwise there is the potential for the avatar to deviate from the user object. Instead, we can just perform this operation on the frontend in an avatar component.

Reply:
How would they deviate? Well have to run the operation for every avatar object there could be 100

Reply:
The deviation could happen when a user's `username` changes after the avatar has been created. It is a little inconvenient that we will have to perform that operation for all of a groups users but even in that worst case I don't think it would take more then a millisecond. Generally you only want to denormalize like that when the operation is highly expensive because data deviations are dangerous.

Reply:
Also, we will probably need to calculate it for non-users anyway so I don't think we can really avoid it by doing it here.

Reply:
Okay, do we want users to change usernames though?

Reply:
I don't see why we would want to block changing them.

Reply:
I don't see why we would want to block changing them.